include/dlaf/matrix/copy_tile.h

```diff
@@ -46,6 +47,16 @@ struct CopyTile<T, Device::CPU, Device::GPU> {
     DLAF_CUDA_CALL(cudaMemcpy2D(destination.ptr(), destination.ld() * sizeof(T), source.ptr(),
                                 source.ld() * sizeof(T), m * sizeof(T), n, cudaMemcpyDefault));
   }
+
+  static void call(const matrix::Tile<const T, Device::CPU>& source,
+                   const matrix::Tile<T, Device::GPU>& destination, cudaStream_t stream) {
+    SizeType m = source.size().rows();
```

Comment:
What about a check for compatibility between source and destination (in terms of sizes)?

I always struggle finding a general guideline about this, because who should check? the caller (like we are doing in `copy` now), the one which is responsible for the action (e.g. this one I would say), or both?

It's more a philosophic question than a comment...look forward to your thoughts!